ztp/siteconfig-generator/siteConfig/siteConfigBuilder.go

```diff
@@ -478,3 +485,84 @@ func (scbuilder *SiteConfigBuilder) splitYamls(yamls []byte) ([][]byte, error) {
 
 	return resources, nil
 }
+
+func filterExtraManifests(dataMap map[string]interface{}, filter *Filter) (map[string]interface{}, error) {
+	// return if there's no filter initialized
+	if filter == nil {
+		return dataMap, nil
+	}
+
+	// check if there is an intersection in filenames
+	if filter.Include != nil && filter.Exclude != nil && len(filter.Exclude) > 0 && len(filter.Include) > 0 {
+		m := make(map[string]bool)
+		var fileExistsInBothLists []string
+
+		for _, includeFileName := range filter.Include {
+			m[includeFileName] = true
+		}
+
+		for _, excludeFileName := range filter.Exclude {
+			if _, ok := m[excludeFileName]; ok {
+				fileExistsInBothLists = append(fileExistsInBothLists, excludeFileName)
+			}
+		}
+
+		if len(fileExistsInBothLists) > 0 {
+			stringFiles := strings.Join(fileExistsInBothLists, ",")
+			errStr := fmt.Sprintf("Can not have same file names in both exclude and include lists. Files %s are present in both include and exclude lists", stringFiles)
+			return dataMap, errors.New(errStr)
+		}
+	}
+
+	inclusionDefaultInclude := "include"
+	inclusionDefaultExclude := "exclude"
+	// default value is include. treat use of `exclude` as an advanced feature
+	if filter.InclusionDefault == nil {
+		filter.InclusionDefault = &inclusionDefaultInclude
+	} else if strings.EqualFold(*filter.InclusionDefault, inclusionDefaultExclude) {
+		filter.InclusionDefault = &inclusionDefaultExclude
+	} else if strings.EqualFold(*filter.InclusionDefault, inclusionDefaultInclude) {
+		filter.InclusionDefault = &inclusionDefaultInclude
```

Comment:
Rather than doing string comparisons again below (line 552), you could probably use a boolean internally like `excludeAllByDefault`.

Reply:
added a bool!